### ISSUE TITLE
extension causes terminal problem

### ISSUE THREAD

**Issue by AdvancedBot** [GitLiveApp/GitLive#163]
Type: <b>Bug</b>

after bisec i found it is teamhub

Extension version: 18.0.5
VS Code version: Cursor 0.41.1 (10e3716b510a5d447c1a07b2edd37a352bbea4d0, 2024-09-18T07:38:06.522Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|13th Gen Intel(R) Core(TM) i9-13900HX (32 x 2419)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|13.70GB (6.42GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by AdvancedBot** [comment]
git live extension

**Comment by nbransby** [comment]
What's the problem exactly?

**Comment by omril1** [comment]
I did bisect as well in vscode to see why I had no intellisense and general slowness, it pointed to this extension, disabling it made vscode much more responsive and intellisense is working again.
Is there a way to see what was changed in the latest release?

**Comment by nbransby** [comment]
@omril1 here are the details of the release: https://blog.git.live/gitlive-18.0-merge-conflict-detection-now-flags-conflicting-changes-in-the-repository-view

I suspect it is the merge conflict detection, either in the repository view or in the editor. Do you have the repository view open when the issue occurs? If you change the left view option for their branches from your working tree to base branch, does that fix the issue?
![17275656047735193756311298329409](https://github.com/user-attachments/assets/62150a52-f3f9-49b9-b4a0-1c21a279fce6)

**Comment by AdvancedBot** [comment]
> What's the problem exactly?

![image](https://github.com/user-attachments/assets/813c07c9-b343-4e95-8ca9-c11f821aec18)
The reactivating terminal in the bottom left corner keeps spinning.

**Comment by AdvancedBot** [comment]
When I disable this extension, it won't happen again.

**Comment by MKlblangenois** [comment]
Same issue on my side, Git live give long time and execution error for terminal, git and other extension error (like Prettier taking 7-8s per task). Disabled Git live solve issue

**Comment by omril1** [comment]
After waiting a while and letting the extension to load more the issue is resolved, but I ended up removing it anyway since it shows too much info on hover and there is no option to disable this as far as I can see (there are barely any options at all)

**Comment by nbransby** [comment]
@omril1 you can disable that via right-click in the gutter.

We are waiting on the vscode team to control the hover verbosity: https://github.com/GitLiveApp/GitLive/issues/150

**Comment by nbransby** [comment]
@omril1 @MKlblangenois are you able to confirm if you are using vscode insiders or stable?

**Comment by MKlblangenois** [comment]
@nbransby I use the stable version. I could try with the insider version if you need reproduction ?

<img width="331" alt="image" src="https://github.com/user-attachments/assets/dde9928d-1412-47f2-bc4a-de034d41ef04">

**Comment by MKlblangenois** [comment]
@nbransby with the Insider version, I can't reproduce the issue, it seems to work better

**Comment by nbransby** [comment]
Thanks @MKlblangenois

**Comment by nbransby** [comment]
Version 18.0.6 of GitLive has now been released which should fix these issues, please let us know if you are still experiencing the issue after updating the extension

**Comment by xin053** [comment]
Version 18.0.6 of GitLive still have this problem

**Comment by nbransby** [comment]
@xin053 can you give more details on the issue you're experiencing exactly?

**Comment by MP3Martin** [comment]
> Version 18.0.6 of GitLive still have this problem

For me, it causes a memory leak of 10+ MB per second

(by not doing anything, just having the extension enabled)

**Comment by nbransby** [comment]
@MP3Martin do you have any particularly large files in your git repository?

**Comment by MP3Martin** [comment]
> @MP3Martin do you have any particularly large files in your git repository?

no files/tabs open, just the extension enabled
disabling the extension fixes my issue

I made the comment just to share some info, i don't need to resolve the issues because i have disabled the extension (i don't use its features that often). My issue might also be caused by an incompatibility with a different extension.

**Comment by nbransby** [comment]
> I made the comment just to share some info,

Thanks. Did the issue occur on an open source repo you are able to share by any chance? Also if you could share the list of extensions you have installed that would help a lot too.

**Comment by MP3Martin** [comment]
> > I made the comment just to share some info,
> 
> Thanks. Did the issue occur on an open source repo you are able to share by any chance? Also if you could share the list of extensions you have installed that would help a lot too.

Sorry, I've just realized what this extension is for. The issue occurs even with a new window of VSCode that doesn't have any folder/file/repo open.

My extension list is very long.

I've now tried to disable every single extension except GitLive and the issue still happens. I am using default GitLive setting (i don't think there are any)

**Comment by MP3Martin** [comment]
> > I made the comment just to share some info,
> 
> Thanks. Did the issue occur on an open source repo you are able to share by any chance? Also if you could share the list of extensions you have installed that would help a lot too.

Sorry, I've just realized what this extension is for. The issue occurs even with a new window of VSCode that doesn't have any folder/file/repo open.

My extension list is very long.

I've now tried to disable every single extension except GitLive, relaunched VSCode and the issue still happens. I am using default GitLive setting (i don't think there are any)

**Comment by nbransby** [comment]
> The issue occurs even with a new window of VSCode that doesn't have any folder/file/repo open.

That's wild. Do you also see this reactivating terminals message?

Is there anything useful in the output window, under Extension Host or Shared maybe?

**Comment by nbransby** [comment]
Also if you are able, could you run this command?
<img width="607" alt="image" src="https://github.com/user-attachments/assets/a86caf14-681f-4c25-97f3-e17a70921afb">

**Comment by MP3Martin** [comment]
If it try to run the command quickly, then it maybe works. But if i run it like 10 seconds after opening vscode, the extension is already frozen

![image](https://github.com/user-attachments/assets/edf838ae-2c82-4ec7-a44b-c2f013dbb37e)


Edit: if it was supposed to give me something after pressing enter, then it didn't work.

**Comment by nbransby** [comment]
> Edit: if it was supposed to give me something after pressing enter, then it didn't work.

Nvm, It was.

The fact you get it even on the welcome page means it must be something very fundamental that's not working as very little code executes before you have opened a workspace with a git repo in it.

Could you share the logs mentioned in this [issue](https://github.com/angular/vscode-ng-language-service/issues/1087), namely:

Console output (Go to toolbar --> View --> Output, and copy the contents)
Log file (Path is printed to the console output at startup)

**Comment by nbransby** [comment]
Also this would be super useful if you have the time https://github.com/microsoft/vscode/wiki/performance-issues#profile-the-running-extensions

Apologies for all the requests, we've had no luck reproducing this issue and you seem like one of the very few people able to reliably reproduce it

**Comment by MP3Martin** [comment]
> > Edit: if it was supposed to give me something after pressing enter, then it didn't work.
> 
> Nvm, It was.
> 
> The fact you get it even on the welcome page means it must be something very fundamental that's not working as very little code executes before you have opened a workspace with a git repo in it.
> 
> Could you share the logs mentioned in this [issue](https://github.com/angular/vscode-ng-language-service/issues/1087), namely:
> 
> Console output (Go to toolbar --> View --> Output, and copy the contents) Log file (Path is printed to the console output at startup)

I don't have a "GitLive" option in this "Output" dropdown
![image](https://github.com/user-attachments/assets/d1e742e2-4f9f-449f-a1bb-9584dc80a33c)
(the image doesn't contain all of the dropdown options)

The only path i could find is this:
![image](https://github.com/user-attachments/assets/a8b69865-784d-443e-a99d-6ee93a994a81)
(When i select "Window" from the dropdown)

This is the content of the automatically generated .cpuprofile :
https://gist.githubusercontent.com/MP3Martin/1669bc783b1715bda57cf6d664591efe/raw/4b5e9603e02b0444a6e9cf683f57927ad3e1810d/exthost-7fef10.cpuprofile

**Comment by MP3Martin** [comment]
> Also this would be super useful if you have the time https://github.com/microsoft/vscode/wiki/performance-issues#profile-the-running-extensions
> 
> Apologies for all the requests, we've had no luck reproducing this issue and you seem like one of the very few people able to reliably reproduce it

This one is manually recorded:
https://gist.githubusercontent.com/MP3Martin/1669bc783b1715bda57cf6d664591efe/raw/dca1e33ae7a5ff600a2103138faee60fcb042150/CPU-20241011T080009.134Z.cpuprofile.txt

**Comment by MP3Martin** [comment]
Here is also a video of the growing memory usage 

https://github.com/user-attachments/assets/7fc83b79-1860-49b4-b223-2e48fe499cca

**Comment by MP3Martin** [comment]
Here is also a video of the growing memory usage:
(not sped up)

https://github.com/user-attachments/assets/07ea04e2-0026-4a07-8a96-057df0a0fded

**Comment by nbransby** [comment]
> This one is manually recorded:

was this recorded whilst the memory was going up? There is no cpu usage by GtiLive in that profile

**Comment by MP3Martin** [comment]
> > This one is manually recorded:
> 
> was this recorded whilst the memory was going up? There is no cpu usage by GtiLive in that profile

I wanted to comment to #165, but someone said that it was a duplicate of this issue (#163), so i made my comment here

My problem is memory going up and the extension being frozen, and not a cpu usage problem

**Comment by nbransby** [comment]
Got it, from my understanding we should still see the some activity on the cpu profile from gitlive though if its code is executing to allocate all that memory.

Does the issue still happen if you install the previous version, 17.1.4?

<img width="546" alt="image" src="https://github.com/user-attachments/assets/a00888a7-39c4-4c46-8427-4789ee107ff3">

**Comment by MP3Martin** [comment]
17.1.4 works fine

18.0.5 and 18.0.6 have the issue

**Comment by nbransby** [comment]
what about 18.0.0...

https://marketplace.visualstudio.com/_apis/public/gallery/publishers/TeamHub/vsextensions/teamhub/18.0.0/vspackage

**Comment by MP3Martin** [comment]
> what about 18.0.0...
> 
> https://marketplace.visualstudio.com/_apis/public/gallery/publishers/TeamHub/vsextensions/teamhub/18.0.0/vspackage

![image](https://github.com/user-attachments/assets/48513f75-5940-4d97-86e2-a33d160578ed)
still the same issue

**Comment by nbransby** [comment]
> Could you share the logs mentioned in this [issue](https://github.com/angular/vscode-ng-language-service/issues/1087), namely:
> 
> Console output (Go to toolbar --> View --> Output, and copy the contents)
>  Log file (Path is printed to the console output at startup)

Could you share these?

**Comment by MP3Martin** [comment]
> Could you share these?

[I don't have a "GitLive" option in the "Output" dropdown](https://github.com/GitLiveApp/GitLive/issues/163#issuecomment-2406779461)

Which dropdown option's content should i share?

**Comment by nbransby** [comment]
View --> Output
also the log file (Path is printed to the console output at startup)
Also Help -> Toggle Developer Tools -> Console - if there anything in there thanks

**Comment by MP3Martin** [comment]
![image](https://github.com/user-attachments/assets/ac218cb1-6688-46d3-bfaa-ad6f5edd135a)
![image](https://github.com/user-attachments/assets/88b46daa-84a9-4d96-9db7-3e2ea16335cb)

**Comment by nbransby** [comment]
Could you upload this file?
<img width="1042" alt="image" src="https://github.com/user-attachments/assets/f93ef27e-3d54-45fb-aa76-e25fe10565a8">

**Comment by MP3Martin** [comment]
I have already done that here: https://github.com/GitLiveApp/GitLive/issues/163#issuecomment-2406779461
(the last link in that comment)

**Comment by nbransby** [comment]
That appears to be a different CPU profile to the one mentioned in the logs, didn't you say that one was manually generated?

**Comment by MP3Martin** [comment]
> That appears to be a different CPU profile to the one mentioned in the logs, didn't you say that one was manually generated?

Sorry, i have now edited the comment

auto: https://gist.githubusercontent.com/MP3Martin/1669bc783b1715bda57cf6d664591efe/raw/4b5e9603e02b0444a6e9cf683f57927ad3e1810d/exthost-7fef10.cpuprofile

manual: https://gist.githubusercontent.com/MP3Martin/1669bc783b1715bda57cf6d664591efe/raw/dca1e33ae7a5ff600a2103138faee60fcb042150/CPU-20241011T080009.134Z.cpuprofile.txt

**Comment by nbransby** [comment]
@MP3Martin thanks, that was useful, can you also reproduce the issue in vscode insiders?

**Comment by MP3Martin** [comment]
Hello, i can't reproduce the issue in both insiders and vscodium. (with settings sync enabled / profile imported to download the config from normal vscode)

I have also tried to create a new empty profile in my normal VSCode installation (the one where the issue happens), installed just the GitLive extension and i couldn't reproduce the issue.

Then, in the standard version of VSCode, I've created a duplicate (called _"Default (Copy)"_) of the problematic profile (called _"Default"_)
![image](https://github.com/user-attachments/assets/373ac032-0663-4bc8-96a3-8bddb728d964)

and the issue was gone.

I don't know why, but creating a duplicate of the current profile and setting the duplicate as the default profile fixes the issue.

**Comment by nbransby** [comment]
Thanks for trying all that and the detailed explanations, it's much appreciated.

Did GitLive ask you to sign in again on the copy of the default profile? Were you signed in previously with the default profile, for example when you installed 17.1.4 and it was working?

**Comment by MP3Martin** [comment]
> Did GitLive ask you to sign in again on the copy of the default profile?

- This is what it shows:
![image](https://github.com/user-attachments/assets/e036287b-395c-4247-9e03-bb23e84488e3)
- It didn't show a notification to sing in

> Were you signed in previously with the default profile, for example when you installed 17.1.4 and it was working?
- No

**Comment by MP3Martin** [comment]
So, this is gonna be a little embarrassing

I've just realized that i have been using gitlive and gitlens together

I initially thought that the second extension had features like the git graph, and that this extension included features like this:
![image](https://github.com/user-attachments/assets/0e061c9b-5bf8-45bd-8e85-9ea0385ba862)


However, I've now realized that all those features are from the second extension, and **I misunderstood the purpose of this extension**. If I'm the only one having my **specific** issue _(the quick memory usage growing)_, then you don't have to try to solve my issue anymore, because I've basically never used this extension, it was just installed without any purpose.

I am sorry if i wasted your time

Have a nice day and good luck with everything

**Comment by nbransby** [comment]
Thats ok, I don't believe you are the only one, thanks for your help!

**Comment by nbransby** [comment]
@MP3Martin 
If you are willing to do one last thing - could you share an export of the offending profile to support@gitl.ive to help us reproduce the issue?

**Comment by MP3Martin** [comment]
> @MP3Martin 
> If you are willing to do one last thing - could you share an export of the offending profile to support@gitl.ive to help us reproduce the issue?

But the problem is that when i export and import the profile, the issue doesn't occur

Should i still send the profile?

**Comment by nbransby** [comment]
In that case there is no need

**Comment by nbransby** [comment]
Could you confirm you still get these issues in the latest version (18.0.8)?

**Comment by AdvancedBot** [comment]
> Could you confirm you still get these issues in the latest version (18.0.8)?

The problem has been solved, thank you for your help!

**Comment by xin053** [comment]
The problem has been solved
